version: '3'
services:
  reverse-proxy:
    image: traefik
    command: --api.insecure=true --providers.docker --providers.docker.exposedByDefault=false
    restart: on-failure
    ports:
      - "80:80"
      - "8080:8080"
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock

  redis:
    image: redis:alpine
    restart: on-failure
    command: --appendonly yes
    volumes:
      - redis_volume:/data

  mongo:
    image: mongo
    restart: on-failure
    volumes:
      - ./conf/mongo-init.js:/docker-entrypoint-initdb.d/mongo-init.js:ro
      - mongo_volume:/data/db
    environment:
      - MONGO_INITDB_ROOT_USERNAME=root
      - MONGO_INITDB_ROOT_PASSWORD=example
      - MONGO_INITDB_DATABASE=admin

  swagger-ui:
    image: swaggerapi/swagger-ui
    restart: on-failure
    volumes:
      - ./conf/api.yaml:/usr/share/nginx/html/swagger/api.yaml:ro
    environment:
      API_URL: swagger/api.yaml
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.swagger-ui.rule=Host(`swagger-ui.vagrantbox.local`)"
      - "traefik.http.services.swagger-ui.loadbalancer.server.port=8080"

volumes:
  mongo_volume:
  redis_volume:
